solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -2983,6 +2983,57 @@ impl RemoveParaIdsWithNoCredits for RemoveParaIdsWithNoCreditsImpl {
     }
 }
 
+pub struct GetCoreAllocationConfigurationImpl;
+
+impl Get<Option<CoreAllocationConfiguration>> for GetCoreAllocationConfigurationImpl {
+    fn get() -> Option<CoreAllocationConfiguration> {
+        let current_block = System::block_number();
+
+        let active_config = runtime_parachains::configuration::ActiveConfig::<Runtime>::get();
+        let mut pending_configs =
+            runtime_parachains::configuration::PendingConfigs::<Runtime>::get();
+
+        // We cannot use `<Runtime as pallet_session::Config>::ShouldEndSession` as it is not a pure query and writes
+        // to storage in certain conditions (for example, when this is invoked on genesis.) which makes current_slot =
+        // block - 1 instead of being equal.
+        let should_end_session = Babe::should_epoch_change(current_block);
```

Comment:
This is wrong, babe does not work with epochs but slots. So instead of current_block, please fetch the current slot from babe

https://github.com/moondance-labs/tanssi/blob/645538b4dbcb54de033489e8db2c5a18621cbb5c/solo-chains/runtime/starlight/src/lib.rs#L2864

Reply:
@girazoki This has been removed in latest version. It turns out `should_end_session` call only make sense to be used inside `runtime_api`, otherwise the answer will always be false when used by collator assignment pallet on `initializer_on_new_session` as well as in the extrinsics.

Reply:
@girazoki That is exactly what I did. :)

Reply:
Thanks!

Reply:
*fixed*